client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoPreparedStatement.java

```diff
@@ -132,12 +134,22 @@ public ResultSet executeQuery()
             throws SQLException
     {
         requireNonBatchStatement();
-        if (!super.execute(getExecuteSql(statementName, toValues(parameters)))) {
+
+        if (!super.execute(getSqlToExecute())) {
```

Comment:
There are more places where `getExecuteSql()` is invoked (`executeBatch()`, `execute()`, `executeLargeUpdate()`).  Those call sites also need to be updated.  Maybe get rid of `getSqlToExecute()` and move the logic to `getExecuteSql()` and add a `useLegacyPreparedStatements`, then the compiler will help you find all places that need to be updated.

Reply:
Because `getExecuteSql()` is static it cannot check whether to use execute immediate or not. But let me think of a way around this.

Reply:
Code updated, and tests added to cover these new scenarios.